### DISCUSSION TITLE
ArgumentNullException being thrown in NumericValidationBehavior

### DISCUSSION THREAD

**Discussion by COMH-AIQ** [CommunityToolkit/Maui#1140] [General]
I have a MVVM app targeting Windows and in my viewmodel I have a nullable decimal property bound to an Entry control. I'm using the NumericValidationBehavior and it's throwing an exception when the decimal value is null, in the validator `ValidateAsync` call because the 'value' parameter is null.

```
	protected override ValueTask<bool> ValidateAsync(string? value, CancellationToken token)
	{
		ArgumentNullException.ThrowIfNull(value);
		
		if (!(double.TryParse(value, out var numeric)
				&& numeric >= MinimumValue
				&& numeric <= MaximumValue))
		{
			return new ValueTask<bool>(false);
		}
```
		
Looks like that might be a bug? There's no need to throw there because the TryParse can accept null, and will just return "false" and return the ValueTask. Since my application of this validator is to allow the user to supply a value when it is missing, it will have `null` as a value and when it binds it throws the exception and the application just crashes and disappears.

**Comment by bijington** [comment]
I would agree that it is likely a bug. If setting

```csharp
builder.UseMauiCommunityToolkit(options =>
{
    options.SetShouldSuppressExceptionsInBehaviors(true);
});
```
doesn't solve the issue (full details at  https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/get-started?tabs=CommunityToolkitMaui#using-the-nuget-packages) then the bug is that it doesn't abide by that option.

Are you able to test whether enabling that option makes a different? And if it doesn't are you able to open a bug please?

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
